homeassistant/components/vicare/button.py

```diff
@@ -0,0 +1,150 @@
+"""Viessmann ViCare sensor device."""
+from __future__ import annotations
+
+from contextlib import suppress
+from dataclasses import dataclass
+import logging
+
+from PyViCare.PyViCareUtils import (
+    PyViCareInvalidDataError,
+    PyViCareNotSupportedFeatureError,
+    PyViCareRateLimitError,
+)
+import requests
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ENTITY_CATEGORY_CONFIG
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ViCareRequiredKeysMixin
+from .const import DOMAIN, VICARE_API, VICARE_DEVICE_CONFIG, VICARE_NAME
+
+_LOGGER = logging.getLogger(__name__)
+
+BUTTON_DHW_ACTIVATE_ONETIME_CHARGE = "activate_onetimecharge"
+
+
+@dataclass
+class ViCareButtonEntityDescription(ButtonEntityDescription, ViCareRequiredKeysMixin):
+    """Describes ViCare button sensor entity."""
+
+
+BUTTON_DESCRIPTIONS: tuple[ViCareButtonEntityDescription, ...] = (
+    ViCareButtonEntityDescription(
+        key=BUTTON_DHW_ACTIVATE_ONETIME_CHARGE,
+        name="Activate one-time charge",
+        icon="mdi:shower-head",
+        entity_category=ENTITY_CATEGORY_CONFIG,
+        device_class=ButtonDeviceClass.RESTART,
+        value_getter=lambda api: api.activateOneTimeCharge(),
+    ),
+)
+
+
+def _build_entity(name, vicare_api, device_config, sensor):
```

Comment:
Looks like this function isn't needed.

Reply:
it is used. `_build_entity` is used in `async_setup_entry`

Reply:
Yes, it's used, but it isn't needed since it just instantiates the entity class. We can do that directly in `async_setup_entry`.

Reply:
agreed